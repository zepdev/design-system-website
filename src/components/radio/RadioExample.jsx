import React, { useState } from "react"
import Radio from "@zlab-de/zel-react/Radio"
import RadioGroup from "@zlab-de/zel-react/RadioGroup"

function RadioExample() {
  const [radio, setRadio] = useState("a")
  const [radioInline, setRadioInline] = useState("a")

  return (
    <div>
      <RadioGroup title="Select One">
        <Radio
          value="a"
          id="r1"
          label="a"
          name="radioGroup"
          checked={radio === "a"}
          onChange={e => setRadio(e.target.value)}
        />
        <Radio
          value="b"
          id="r2"
          name="radioGroup"
          label="b"
          checked={radio === "b"}
          onChange={e => setRadio(e.target.value)}
        />
      </RadioGroup>
      <RadioGroup title="Inline Example">
        <Radio
          variant="inline"
          value="a"
          label="inline a"
          id="r1Inline"
          name="radioGroup2"
          checked={radioInline === "a"}
          onChange={e => setRadioInline(e.target.value)}
        />
        <Radio
          variant="inline"
          value="b"
          label="inline b"
          id="r2Inline"
          name="radioGroup2"
          checked={radioInline === "b"}
          onChange={e => setRadioInline(e.target.value)}
        />
      </RadioGroup>
    </div>
  )
}

export default RadioExample
