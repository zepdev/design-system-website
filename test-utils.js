import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/core/styles'
import theme from './src/theme'

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
