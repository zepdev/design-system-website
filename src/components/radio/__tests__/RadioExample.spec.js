import React from 'react'
import { render } from '../../../../test-utils'
import RadioExample from '../RadioExample'

describe('RadioExample', () => {
  it('renders correctly', () => {
    const component = render(<RadioExample />)
    expect(component).toMatchSnapshot()
  })
})
