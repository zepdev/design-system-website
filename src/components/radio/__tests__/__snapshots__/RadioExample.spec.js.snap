// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioExample renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <fieldset
          aria-labelledby="radio_Select One"
          class="zep-typo--body-default"
          role="group"
        >
          <legend
            id="radio_Select One"
          >
            Select One
          </legend>
          <div
            class="zep-radio"
          >
            <input
              checked=""
              class="zep-radio__input"
              id="r1"
              name="value 1"
              type="radio"
              value="a"
            />
            <label
              class="zep-radio__label"
              for="r1"
            >
              value 1
            </label>
          </div>
          <div
            class="zep-radio"
          >
            <input
              class="zep-radio__input"
              id="r2"
              name="value 2"
              type="radio"
              value="b"
            />
            <label
              class="zep-radio__label"
              for="r2"
            >
              value 2
            </label>
          </div>
        </fieldset>
        <fieldset
          aria-labelledby="radio_Inline Example"
          class="zep-typo--body-default"
          role="group"
        >
          <legend
            id="radio_Inline Example"
          >
            Inline Example
          </legend>
          <div
            class="zep-radio zep-radio--inline"
          >
            <input
              checked=""
              class="zep-radio__input"
              id="r1Inline"
              name="value 1"
              type="radio"
              value="a"
            />
            <label
              class="zep-radio__label"
              for="r1Inline"
            >
              value 1
            </label>
          </div>
          <div
            class="zep-radio zep-radio--inline"
          >
            <input
              class="zep-radio__input"
              id="r2Inline"
              name="value 2"
              type="radio"
              value="b"
            />
            <label
              class="zep-radio__label"
              for="r2Inline"
            >
              value 2
            </label>
          </div>
        </fieldset>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <fieldset
        aria-labelledby="radio_Select One"
        class="zep-typo--body-default"
        role="group"
      >
        <legend
          id="radio_Select One"
        >
          Select One
        </legend>
        <div
          class="zep-radio"
        >
          <input
            checked=""
            class="zep-radio__input"
            id="r1"
            name="value 1"
            type="radio"
            value="a"
          />
          <label
            class="zep-radio__label"
            for="r1"
          >
            value 1
          </label>
        </div>
        <div
          class="zep-radio"
        >
          <input
            class="zep-radio__input"
            id="r2"
            name="value 2"
            type="radio"
            value="b"
          />
          <label
            class="zep-radio__label"
            for="r2"
          >
            value 2
          </label>
        </div>
      </fieldset>
      <fieldset
        aria-labelledby="radio_Inline Example"
        class="zep-typo--body-default"
        role="group"
      >
        <legend
          id="radio_Inline Example"
        >
          Inline Example
        </legend>
        <div
          class="zep-radio zep-radio--inline"
        >
          <input
            checked=""
            class="zep-radio__input"
            id="r1Inline"
            name="value 1"
            type="radio"
            value="a"
          />
          <label
            class="zep-radio__label"
            for="r1Inline"
          >
            value 1
          </label>
        </div>
        <div
          class="zep-radio zep-radio--inline"
        >
          <input
            class="zep-radio__input"
            id="r2Inline"
            name="value 2"
            type="radio"
            value="b"
          />
          <label
            class="zep-radio__label"
            for="r2Inline"
          >
            value 2
          </label>
        </div>
      </fieldset>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
